library/src/SPSApplication.ts

```diff
@@ -1,97 +1,263 @@
+            clearInterval(this.statsTimer);
+        }
+    }
+
+    updateStatValue(name: string, value: number) {
+        if (!ENABLE_METRICS) {
+            return;
+        }
+
+        if (value == null) {
+            return;
+        }
+
+        const statDescription = SupportedStats[name];
+        if (!statDescription) {
+            console.log(`unknown stat ${name}`);
+            return;
+        }
+
+        if (!this.sessionData) {
+            console.log(`no sessiondata`);
+            return;
+        }
+
+        if (!this.sessionData.stats) {
+            this.sessionData.stats = {};
+        }
+
+        if (this.sessionData.stats[name]) {
+            this.sessionData.stats[name].value = value;
+        } else {
+            this.sessionData.stats[name] = {
+                description: statDescription,
+                value: value
+            };
+        }
+    }
+
+    onSessionStats(aggregatedStats: AggregatedStats) {
+        if (!ENABLE_METRICS) {
+            return;
+        }
+
+        if (!this.sessionData) {
+            return;
+        }
+
+        this.sessionData.lastSent = Date.now();
+
+        // if sessionData is defined we can assume the session is active
+        if (aggregatedStats.inboundVideoStats) {
+            const videoStats = aggregatedStats.inboundVideoStats;
+            this.updateStatValue("video_width", videoStats.frameWidth);
+            this.updateStatValue("video_height", videoStats.frameHeight);
+            this.updateStatValue("video_bitrate", videoStats.bitrate);
+            this.updateStatValue("video_dropped", videoStats.framesDropped);
+            this.updateStatValue("video_packets_lost", videoStats.packetsLost);
+            // rtt?
+            this.updateStatValue("video_fps", videoStats.framesPerSecond);
+            this.updateStatValue("video_pli_count", videoStats.pliCount);
+            this.updateStatValue("video_keyframes", videoStats.keyFramesDecoded);
+            this.updateStatValue("video_nack_count", videoStats.nackCount);
+            this.updateStatValue("video_freeze_count", videoStats.freezeCount);
+            this.updateStatValue("video_jitter", videoStats.jitter);
+            this.updateStatValue("video_frame_count", videoStats.framesReceived);
+        }
+        if (aggregatedStats.inboundAudioStats) {
+            const audioStats = aggregatedStats.inboundAudioStats;
+            this.updateStatValue("audio_bitrate", audioStats.bitrate);
+        }
+
+        // POST the stats
+        this.httpPost(this.sessionData.stats);
+    }
+
+    async httpPost(data: any) {
+        for (const prop in data) {
+            if (typeof data[prop] === 'number') {
+                data[prop] = data[prop].toString();
+            }
+        }
+        const post_data = {
+            id: this.sessionData.id,
+            metrics: data
+        };
+
+        const response = await fetch(`http://${window.location.hostname}:8000/stats`, {

```

Comment:
Consider swapping out this hardcoded stats url with an env variable, like the websocket url.